File: Ejercicio1/Carrera.java
package Ejercicio1;
import java.util.ArrayList;

public abstract class Carrera {
    private ArrayList<Bicicleta> bicicletas; //Array de bicicletas

    public Carrera(ArrayList<Bicicleta> bicicletas) {
        this.bicicletas = bicicletas;
    }

    public ArrayList<Bicicleta> getBicicletas() {
        return bicicletas;
    }

    public void setBicicletas(ArrayList<Bicicleta> bicicletas) {
        this.bicicletas = bicicletas;
    }

    public abstract void empezarCarrera() throws InterruptedException;
}

File: Ejercicio2/Carrera.py
from abc import ABC, abstractmethod
from Bicicleta import Bicicleta
from typing import List

class Carrera(ABC):
    def __init__(self, bicicletas: List['Bicicleta']):
        self.bicicletas = bicicletas

    # Es un getter
    def getBicicletas(self):
        return self._bicicletas

    # Es un setter
    def setBicicletas(self, bicicletas: List['Bicicleta']):
        self._bicicletas = bicicletas

    @abstractmethod
    def empezar_carrera(self):
        pass

    def eliminar_bicicleta(self, pos: int):
        del self.bicicletas[pos]
        